alexandrie/src/main.rs

```diff
@@ -83,14 +86,22 @@ async fn run() -> Result<(), Error> {
 
     info!("running database migrations");
     #[rustfmt::skip]
-    state.repo.run(|conn| embedded_migrations::run(conn)).await
+        state.repo.run(|conn| embedded_migrations::run(conn)).await
         .expect("migration execution error");
 
     let mut app = tide::with_state(Arc::new(state));
 
     info!("setting up request logger middleware");
     app.middleware(RequestLogger::new());
 
+    //handle when response error,set error message into body.
+    app.middleware(After(|mut res: Response| async {
+        if let Some(err) = res.error() {
+            res.set_body(format!("Error: {}", err.to_string()));
```

Comment:
After taking a look to the [**Alternative Registries documentation page in the Cargo book**](https://doc.rust-lang.org/cargo/reference/registries.html#web-api), errors from the registries are supposed to be reported as a JSON payload with a status code of 200.  

And that JSON payload should be of the following shape:
```jsonc
{
    // Array of errors to display to the user.
    "errors": [
        {
            // The error message as a string.
            "detail": "error message text"
        }
    ]
}
```

```suggestion
            let payload = json::json!({
                "errors": [{
                    "details": err.to_string(),
                }]
            });
            res.set_status(200);
            res.set_content_type(tide::http::mime::JSON);
            res.set_body(tide::Body::from_json(&payload).unwrap());
```

Reply:
Yeah, I just take a look taking a look to the [Alternative Registries documentation page in the Cargo book](https://doc.rust-lang.org/cargo/reference/registries.html#web-api).The document suggest use `detail` keyword. 
Has fixed!